plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsRepository.java

```diff
@@ -109,6 +111,30 @@ private HdfsBlobStore createBlobstore(URI blobstoreUri, String path, Settings re
             hadoopConfiguration.set(key, confSettings.get(key));
         }
 
+        Integer replicationFactor = repositorySettings.getAsInt("create_opts.replication", null);
+        int minReplicationFactory = hadoopConfiguration.getInt("dfs.replication.min", 0);
+        int maxReplicationFactory = hadoopConfiguration.getInt("dfs.replication.max", 512);
+        if (replicationFactor != null && replicationFactor < minReplicationFactory) {
+            throw new RepositoryException(
+                metadata.name(),
+                MessageFormat.format(
+                    "Value of create_opts.replication [{0}] must be >= dfs.replication.min [{1}]",
+                    replicationFactor,
+                    minReplicationFactory
+                )
```

Comment:
Fun fact: The RepositoryException extends ElasticsearchException which has message formatting baked in. You can just pass in the arguments as-is. I think the curly braces are fine with being empty as well, so just `{}` should work for the placeholders.

Reply:
Apologies, but I'm not entirely sure if I understand what's the suggested change here. The constructor with message formatting is in `ElasticsearchException`, but `RepositoryException` does not implement those from what I can see.

I've proposed adding vararg to `RepositoryException` constructor. Please let me know if it's OK to do it in this PR.

Alternatively, I can just instantiate `ElasticsearchException` instead, but then it will be wrapped in another `RepositoryException` [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java#L729), which makes the returned json error more complicated - root cause becomes `cannot create blob store` in that case.